src/main/java/com/example/hop_oasis/decoder/ExtractImageName.java

```diff
@@ -0,0 +1,14 @@
+package com.example.hop_oasis.decoder;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class ExtractImageName {
```

Comment:
The class name cannot be an action(verb) it should be a noun. `ImageNameExtractor`